src/kontrol/__main__.py

```diff
@@ -71,6 +75,25 @@ def main() -> None:
     execute(**vars(args))
 
 
+def _check_k_version() -> None:
+    expected_k_version = KVersion.parse(f'v{pyk.K_VERSION}')
+    actual_k_version = k_version()
+
+    if (
+        expected_k_version != actual_k_version
+        and not _compare_versions(expected_k_version, actual_k_version)
+        and not _compare_versions(actual_k_version, expected_k_version)
+    ):
```

Comment:
Please continue moving logic into `_compare_versions` until this condition simplifies to
```suggestion
    if not _compare_versions(expected_k_version, actual_k_version):
```

Then we can focus on simplifying the code in that function.

Reply:
Ok, I just refactored! Please check that the logic is ok. I can refactor the `_compare_versions` function even further to have just one `return True` and `return False`. Let me know.